plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/portforwading/PortForwardingWizardModel.java

```diff
@@ -208,10 +208,26 @@ public void stopPortForwarding() {
 			MessageConsoleStream stream = console.newMessageStream();
 			stream.println("Stopping port-forwarding...");
 			cap.stop();
+			//Wait up to 5 seconds until the forcibly destroyed process really dies.
+			for (int i = 0; i < 50 && hasPortInUse(); i++) {
```

Comment:
Waiting block is moved to a private method.

Sockets are autoclosable. When created in the header () of try block, Java 1.8 adds behind the scene finally which closes created resources.